src/ui/suspense/nodeCaches.ts

```diff
@@ -173,27 +160,157 @@ export const nodeDataCache: Cache<
   },
 });
 
+export const processedNodeDataCache: Cache<
+  [replayClient: ReplayClientInterface, pauseId: PauseId, nodeId: string],
+  NodeInfo | null
+> = createCache({
+  config: { immutable: true },
+  debugLabel: "ProcessedNodeData",
+  getKey: ([replayClient, pauseId, nodeId]) => `${pauseId}:${nodeId}`,
+  load: async ([replayClient, pauseId, nodeId]): Promise<NodeInfo | null> => {
+    if (!pauseId || typeof nodeId !== "string") {
+      return null;
+    }
+    const nodeObject = await objectCache.readAsync(replayClient, pauseId, nodeId, "canOverflow");
+
+    const node = nodeObject?.preview?.node;
+    assert(node, "No preview for node: " + nodeId);
+
+    return {
+      attributes: node.attributes || [],
+      children: node.childNodes ?? [],
+      displayName:
+        node.nodeType === NodeConstants.DOCUMENT_TYPE_NODE
+          ? `<!DOCTYPE ${node.nodeName}>`
+          : node.nodeName.toLowerCase(),
+      hasChildren: !!node.childNodes?.length,
+      id: nodeId,
+      isConnected: node.isConnected,
+      isElement: node.nodeType === NodeConstants.ELEMENT_NODE,
+      namespaceURI: HTML_NS,
+      parentNodeId: node.parentNode,
+      pseudoType: node.pseudoType!,
+      tagName: node.nodeType === Node.ELEMENT_NODE ? node.nodeName : undefined,
+      type: node.nodeType,
+      value: node.nodeValue,
+      isLoadingChildren: false,
+    };
+  },
+});
+
+export const ancestorNodesCache: Cache<
+  [replayClient: ReplayClientInterface, pauseId: PauseId, nodeId: string],
+  NodeInfo[] | null
+> = createCache({
+  config: { immutable: true },
+  debugLabel: "AncestorNodesData",
+  getKey: ([replayClient, pauseId, nodeId]) => `${pauseId}:${nodeId}`,
+  load: async ([replayClient, pauseId, nodeId]): Promise<NodeInfo[] | null> => {
+    if (!pauseId || typeof nodeId !== "string") {
```

Comment:
This is unexpected, given the way the types are defined for this cache:
```js
  [replayClient: ReplayClientInterface, pauseId: PauseId, nodeId: string],
```
Which is wrong?